src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GetTimelineTest.kt

```diff
@@ -123,57 +124,60 @@ class GetTimelineTest : IntegrationTestBase() {
     )
     createActionPlan(prisonNumber, createActionPlanRequest)
 
-    // When
-    val response = webTestClient.get()
-      .uri(URI_TEMPLATE, prisonNumber)
-      .bearerToken(aValidTokenWithViewAuthority(privateKey = keyPair.private))
-      .exchange()
-      .expectStatus()
-      .isOk
-      .returnResult(TimelineResponse::class.java)
-
-    // Then
     val actionPlan = actionPlanRepository.findByPrisonNumber(prisonNumber)
     val goal = actionPlan!!.goals!![0]
-    val actual = response.responseBody.blockFirst()
-    assertThat(actual.events).hasSize(4)
-    val actionPlanCreatedCorrelationId = actual.events[3].correlationId
-    assertThat(actual)
-      .isForPrisonNumber(prisonNumber)
-      .event(1) {
-        it.hasSourceReference("1")
-          .hasEventType(TimelineEventType.PRISON_ADMISSION)
-          .hasPrisonId("MDI")
-          .wasActionedBy("system")
-          .hasNoActionedByDisplayName()
-          .hasNoContextualInfo()
-      }
-      .event(2) {
-        it.hasSourceReference("1")
-          .hasEventType(TimelineEventType.PRISON_TRANSFER)
-          .hasPrisonId("BXI")
-          .wasActionedBy("system")
-          .hasNoActionedByDisplayName()
-          .hasContextualInfo("MDI")
-      }
-      .event(3) {
-        it.hasSourceReference(actionPlan.reference.toString())
-          .hasEventType(TimelineEventType.ACTION_PLAN_CREATED)
-          .hasPrisonId("BXI")
-          .wasActionedBy("auser_gen")
-          .hasActionedByDisplayName("Albert User")
-          .hasNoContextualInfo()
-          .hasCorrelationId(actionPlanCreatedCorrelationId)
-      }
-      .event(4) {
-        it.hasSourceReference(goal.reference.toString())
-          .hasEventType(TimelineEventType.GOAL_CREATED)
-          .hasPrisonId("BXI")
-          .wasActionedBy("auser_gen")
-          .hasActionedByDisplayName("Albert User")
-          .hasContextualInfo(goal.title!!)
-          .hasCorrelationId(actionPlanCreatedCorrelationId)
-      }
+
+    // When
+    await.untilAsserted {
```

Comment:
Because the timeline event is now properly async we need to use awaitility to wait until these assertions pass (awaitility is setup with a 3 second timeout in the integration test superclass)